src/core/qgsstoredexpressionmanager.cpp

```diff
@@ -0,0 +1,180 @@
+/***************************************************************************
+                          qgsstoredexpressionmanager.cpp
+
+ These classes store and control the management and execution of actions
+ associated with a particular Qgis layer. Actions are defined to be
+ external programs that are run with user-specified inputs that can
+ depend on the contents of layer attributes.
+
+                             -------------------
+    begin                : August 2019
+    copyright            : (C) 2019 David Signer
+    email                : david at opengis dot ch
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsstoredexpressionmanager.h"
+
+#include <QDomElement>
+
+QUuid QgsStoredExpressionManager::addStoredExpression( const QString &name, const QString &expression, const QString &tag )
+{
+  Q_UNUSED( tag );
+
+  /* leads to trouble with sip...
+  QgsStoredExpression storedExpression( name, expression );
+  */
+  QgsStoredExpression storedExpression;
+  storedExpression.id = QUuid::createUuid();
+  storedExpression.name = name;
+  storedExpression.expression = expression;
+
+  mStoredExpressions.append( storedExpression );
+
+  return storedExpression.id;
+}
+
+void QgsStoredExpressionManager::removeStoredExpression( const QUuid &id, const QString &tag )
+{
+  Q_UNUSED( tag );
+
+  int i = 0;
+  for ( const QgsStoredExpression &storedExpression : mStoredExpressions )
+  {
+    if ( storedExpression.id == id )
+    {
+      mStoredExpressions.removeAt( i );
```

Comment:
Watch out: you are modifying the container you are looping over!
This only works because you have a break below.
Normally you need to take a (const) copy of the container if modifying the source container in the loop.
So best to leave a short comment why it's allowed to do that here.